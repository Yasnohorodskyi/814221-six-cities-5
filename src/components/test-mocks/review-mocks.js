export const review = {
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`,
  id: 1,
  rating: 4,
  user: {
    photo: `/img/avatar-angelina.jpg`,
    id: 4,
    isPro: false,
    name: `Lena`,
  },
};

export const reviews = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:53.569Z`,
    id: 1,
    rating: 4,
    user: {
      photo: `/img/avatar-angelina.jpg`,
      id: 4,
      isPro: false,
      name: `Lena`,
    },
  },

  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 2,
    rating: 5,
    user: {
      photo: `/img/avatar-angelina.jpg`,
      id: 34,
      isPro: true,
      name: `Masha`,
    },
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:58.569Z`,
    id: 3,
    rating: 3,
    user: {
      photo: `/img/avatar-angelina.jpg`,
      id: 39,
      isPro: false,
      name: `Masha`,
    },
  },
];

export const reviewsBeforeAdapt = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 2,
    rating: 5,
    user: {
      avatar_url: `/img/avatar-angelina.jpg`,
      id: 34,
      is_pro: true,
      name: `Masha`,
    },
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:58.569Z`,
    id: 3,
    rating: 3,
    user: {
      avatar_url: `/img/avatar-angelina.jpg`,
      id: 39,
      is_pro: false,
      name: `Masha`,
    },
  },
];
