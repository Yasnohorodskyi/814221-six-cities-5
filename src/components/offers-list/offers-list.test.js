import React from "react";
import renderer from "react-test-renderer";
import {offers} from "../test-mocks/offers-mocks";
import OffersList from "../offers-list/offers-list";
import {BrowserRouter as Router} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
const noop = () => {};

const styleCardClass = `cities__place-card`;
const styleImgClass = `cities__image-wrapper`;
const widthImg = `260`;
const heightImg = `200`;

describe(`<OffersList /> render`, () => {
  const mockStore = configureStore([]);
  let store = null;
  beforeEach(() => {
    store = mockStore({
      DATA: {
        offersAll: offers,
      },
      USER: {},
      STATE: {
        activeCard: {},
      },
    });
  });

  it(`renders correctly with full information`, () => {
    const offersListComponent = renderer
      .create(
          <Provider store={store}>
            <Router>
              <OffersList
                offers={offers}
                onCardHover={noop}
                styleCardClass={styleCardClass}
                styleImgClass={styleImgClass}
                widthImg={widthImg}
                heightImg={heightImg}
              />
            </Router>
          </Provider>
      )
      .toJSON();

    expect(offersListComponent).toMatchSnapshot();
  });

  it(`renders correctly withought offers`, () => {
    const offersListComponent = renderer
      .create(
          <Provider store={store}>
            <Router>
              <OffersList
                offers={[]}
                onCardHover={noop}
                styleCardClass={styleCardClass}
                styleImgClass={styleImgClass}
                widthImg={widthImg}
                heightImg={heightImg}
              />
            </Router>
          </Provider>
      )
      .toJSON();

    expect(offersListComponent).toMatchSnapshot();
  });
});
