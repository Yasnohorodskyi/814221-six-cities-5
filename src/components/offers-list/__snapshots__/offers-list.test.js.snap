// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OffersList /> render renders correctly with full information 1`] = `
Array [
  <article
    className="cities__place-card place-card"
    dataid={1}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            493
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    dataid={2}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/2"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            323
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          Amazing and Extremely Central Flat
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    dataid={3}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/3"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            218
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/3"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    dataid={4}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/4"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            440
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/4"
          onClick={[Function]}
        >
          Perfectly located Castro
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    dataid={5}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/5"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            134
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/5"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        room
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    dataid={6}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/6"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            226
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/6"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    dataid={7}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/7"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            277
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/7"
          onClick={[Function]}
        >
          Loft Studio in the Central Area
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        hotel
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    dataid={8}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/8"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            129
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/8"
          onClick={[Function]}
        >
          Amazing and Extremely Central Flat
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        room
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    dataid={9}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/9"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            308
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/9"
          onClick={[Function]}
        >
          Canal View Prinsengracht
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        hotel
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    dataid={10}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/10"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            125
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/10"
          onClick={[Function]}
        >
          Penthouse, 4-5 rooms + 5 balconies
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    dataid={11}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/11"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            190
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/11"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    dataid={12}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/12"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            154
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/12"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
]
`;

exports[`<OffersList /> render renders correctly withought offers 1`] = `null`;
